### PR TITLE
 #165022452 Staff should be able to credit a bank account on the app

### PR DESCRIPTION
**Pull request by oleesir**
#### What does this PR do?
Implement credit-account-endpoint
#### Description of Task to be completed?
 ```POST/api/v1/transactions/:accountNumber/credit```
#### How should this be manually tested?
Use postman to test endpoint above with this header:
key:Content-Type value: application/json
#### What are the relevant pivotal tracker stories?
[#165022452](https://www.pivotaltracker.com/story/show/165022452)
#### Any background context you want to add?
N/A
#### Screenshots
N/A

### CODE REVIEW

File: server/src/controllers/Transaction.controller.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    const transactionsLength = transactions.length;
+    const lastId = transactions[transactionsLength - 1].id;
+    const newId = lastId + 1;
+
+    newTransaction.id = newId;
+    newTransaction.createdOn = moment().format();
+    newTransaction.type = 'credit';
+    newTransaction.accountNumber = userAccountNumber;
+    newTransaction.cashier = id;
+    newTransaction.amount = formatAmount(amount);
```
Comment (iverenshaguy) [inline server/src/controllers/Transaction.controller.js:53]: Good job using a util here

File: server/src/controllers/Transaction.controller.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    newTransaction.oldBalance = formatAmount(balance);
+    newTransaction.newBalance = formatAmount(amount) + formatAmount(balance);
+
+    // update account
+    accountToCredit.balance = newTransaction.newBalance;
```
Comment (iverenshaguy) [inline server/src/controllers/Transaction.controller.js:58]: Good job

Comment (iverenshaguy) [reply server/src/controllers/Transaction.controller.js:58]: Please update account to active when payment is made if dormant

File: server/src/controllers/Transaction.controller.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    transactions.push(newTransaction);
+
+    const data = {
+      id: newTransaction.id,
+      accountNumber: newTransaction.accountNumber,
+      amount: newTransaction.amount,
+      cashier: newTransaction.cashier,
+      type: newTransaction.type,
+
```
Comment (iverenshaguy) [inline server/src/controllers/Transaction.controller.js]: Please remove this newline

File: server/src/dummyDb/db.js [hunk 4]
```diff
@@ -57,5 +57,38 @@ export default {
...
+      transactypeType: 'debit',
+      accountNumber: 1123986435,
+      cashier: 3,
+      amount: 6000.00,
+      oldBalance: 10000.00,
+      newBalance: 4000.00
+    },
+    {
+      id: 3,
+      createdOn: 'Thursday, May 30, 2019 5:11 PM',
```
Comment (iverenshaguy) [inline server/src/dummyDb/db.js:84]: Are you sure this is what, `moment().format()` evaluates to?

File: server/src/helpers/formatAmount.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+const formatAmount = value => Number(parseFloat(value).toFixed(2));
```
Comment (iverenshaguy) [inline server/src/helpers/formatAmount.js:8]: Please add a JSDoc

File: server/test/transaction.test.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+          expect(res.body.data).to.have.nested.property('type');
+          expect(res.body.data).to.have.nested.property('accountBalance');
+          expect(res.body.data).to.have.nested.property('amount');
+          expect(res.body).to.have.property('message').eql('Transaction was successful');
+          if (err) return done(err);
+          done();
+        });
+    });
+
+    it('should let a cashier credit an empty amount field', (done) => {
```
Comment (iverenshaguy) [inline server/test/transaction.test.js]:
```suggestion
    it('should not let a cashier credit an account with an empty amount field', (done) => {
```

File: server/src/validations/Transaction.validation.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  * @param {function} next
+  *
+  * @returns {object} status code and error message properties
+  */
+  static validateCreditTransaction(req, res, next) {
+    const { amount } = req.body;
+    if (amount < 1) {
+      return res.status(400).json({
+        status: 400,
+        error: 'Credit transaction cannot be less than 1 Naira'
```
Comment (iverenshaguy) [inline server/src/validations/Transaction.validation.js]:
```suggestion
        error: 'Credit transaction amount cannot be less than 1 Naira'
```

File: server/src/validations/Transaction.validation.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      return res.status(400).json({
+        status: 400,
+        error: 'Transaction amount cannot be empty'
+      });
+    }
+
+    if (!Number(amount)) {
+      return res.status(400).json({
+        status: 400,
+        error: 'Transactions can only contain digits'
```
Comment (iverenshaguy) [inline server/src/validations/Transaction.validation.js]:
```suggestion
        error: 'Transaction amount can only contain digits'
```

File: server/test/transaction.test.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          expect(res.body).to.have.property('status').eql(400);
+          expect(res.body).to.have.property('error').to.eql('Transaction amount cannot be empty');
+          if (err) return done(err);
+          done();
+        });
+    });
+
+    it('should not let a cashier credit an account with negative input', (done) => {
```
Comment (iverenshaguy) [inline server/test/transaction.test.js]:
```suggestion
    it('should not let a cashier credit an account with negative amount', (done) => {
```

File: server/test/transaction.test.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+          expect(res.status).to.equal(400);
+          expect(res.body).to.have.property('status').eql(400);
+          expect(res.body).to.have.property('error').to.eql('Transactions can only contain digits');
+          if (err) return done(err);
+          done();
+        });
+    });
+
+
+    it('should not let a customer credit an account', (done) => {
```
Comment (iverenshaguy) [inline server/test/transaction.test.js:86]: Sweet!!! I'm proud of the dev you are becoming

**Review by iverenshaguy — CHANGES_REQUESTED** [review]
Great job, please review my comments.

File: server/src/controllers/Transaction.controller.js (hunk 1 above)
Comment (oleesir) [reply server/src/controllers/Transaction.controller.js:53]: thanks...

File: server/src/controllers/Transaction.controller.js (hunk 2 above)
Comment (oleesir) [reply server/src/controllers/Transaction.controller.js:58]: done...

File: server/src/controllers/Transaction.controller.js (hunk 3 above)
Comment (oleesir) [reply server/src/controllers/Transaction.controller.js]: done...

File: server/src/dummyDb/db.js (hunk 4 above)
Comment (oleesir) [reply server/src/dummyDb/db.js:84]: done...

File: server/src/helpers/formatAmount.js (hunk 5 above)
Comment (oleesir) [reply server/src/helpers/formatAmount.js:8]: done...

File: server/test/transaction.test.js (hunk 6 above)
Comment (oleesir) [reply server/test/transaction.test.js]: done...

File: server/src/validations/Transaction.validation.js (hunk 7 above)
Comment (oleesir) [reply server/src/validations/Transaction.validation.js]: done...

File: server/src/validations/Transaction.validation.js (hunk 8 above)
Comment (oleesir) [reply server/src/validations/Transaction.validation.js]: done...

File: server/test/transaction.test.js (hunk 9 above)
Comment (oleesir) [reply server/test/transaction.test.js]: done...

File: server/test/transaction.test.js (hunk 10 above)
Comment (oleesir) [reply server/test/transaction.test.js]: thanks to you :))))

Comment (oleesir) [reply server/test/transaction.test.js:86]: thanks

File: server/src/helpers/formatAmount.js (hunk 5 above)
Comment (iverenshaguy) [reply server/src/helpers/formatAmount.js:8]: This should be `string` not `any` please update it in your next PR

File: server/src/helpers/generateNumbers.js [hunk 11]
```diff
@@ -1,7 +1,13 @@
 import db from '../dummyDb/db';
 
 const { accounts } = db;
-
+/**
```
Comment (iverenshaguy) [inline server/src/helpers/generateNumbers.js]: Good job adding a JSDoc. However, this function does not take in any param. Please update it in your next PR. It also doesn't return boolean

File: server/src/helpers/containsAlphabets.js [hunk 12]
```diff
@@ -1,3 +1,10 @@
+/**
+ * @description Function to check that input is not empty, undefined or null
+ *
+ * @param {any} value The data type to be checked
```
Comment (iverenshaguy) [inline server/src/helpers/containsAlphabets.js]: Update to `string`

File: server/src/helpers/generateNumbers.js [hunk 13]
```diff
@@ -1,7 +1,13 @@
...
+ * @description Function to generate 10 Digit Random Number
+ *
+ * @param {any} value The data type to be checked
```
Comment (iverenshaguy) [inline server/src/helpers/generateNumbers.js]: Good job adding a JSDoc. However, this function does not take in any param. It also doesn't return boolean. Please update it.

File: server/src/helpers/isEmpty.js [hunk 14]
```diff
@@ -1,3 +1,10 @@
+/*
```
Comment (iverenshaguy) [inline server/src/helpers/isEmpty.js]: Please fix this. 
```suggestion
/**
```

File: server/src/helpers/validateEmail.js [hunk 15]
```diff
@@ -1,4 +1,12 @@
 /* eslint-disable no-useless-escape */
+
+/*
```
Comment (iverenshaguy) [inline server/src/helpers/validateEmail.js]: Update to 
```suggestion
/**
```

File: server/src/helpers/validateEmail.js [hunk 16]
```diff
@@ -1,4 +1,12 @@
...
+* @description Function to check for a valid email
+*
+* @param {any} value The data type to be checked
```
Comment (iverenshaguy) [inline server/src/helpers/validateEmail.js]:
```suggestion
* @param {string} email The email to be checked
```

**Review by iverenshaguy — CHANGES_REQUESTED** [review]
Please pay more attention to details when writing your JSDocumentation.

**Review by iverenshaguy — APPROVED** [review]
LGTM

**Pull request merged**